guides/advanced/building-sytems.md

```diff
@@ -0,0 +1,349 @@
+   ```
+
+   > **Compatibility Note**: This command is compatible with Debian 11 and 12, and Ubuntu 20.04, 22.04, and anticipated 24.04. Older distributions may require adjustments.
+
+2. **For Other Distributions**
+   Adapt the list of required packages to your system's package manager (e.g., `dnf` for Fedora, `pacman` for Arch Linux). Refer to your distribution's documentation or community for assistance if needed.
+
+> **Tip:** To ensure compatibility, consider using an Ubuntu-based VM if your Linux distribution has significant differences in package names or versions.
+
+#### Windows
+
+- Install [WSL2](https://learn.microsoft.com/en-us/windows/wsl/install) and set up an Ubuntu distribution.
+- Make sure WSL2 is configured to support virtualization.
+
+### Install Erlang and Elixir
+
+If you've already followed the [Nerves Installation guide](https://hexdocs.pm/nerves/installation.html), Erlang and Elixir should be installed on your system. If not, refer to the installation instructions provided in the guide for your operating system.
+
+### Install Nerves Archives
+
+If you've completed the [Nerves Installation guide](https://hexdocs.pm/nerves/installation.html), the Nerves bootstrap archive and local rebar should already be set up. If not, you can install them with:
+
+```bash
+mix archive.install hex nerves_bootstrap
+mix local.rebar
+```
+
+### Cloning the `nerves_systems` Repository
+
+To begin working with Nerves systems, you’ll need to clone the `nerves_systems` repository from GitHub. This repository contains the necessary scripts and configurations for building and maintaining custom Nerves systems.
+
+```bash
+  git clone https://github.com/nerves-project/nerves_systems.git
+  cd nerves_systems
+```
+
+---
+
+## Step 1: Configuring the Build Environment
+
+1. **Copy the Starter Configuration**
+   To begin configuring the environment for building Nerves systems, you need to create a configuration file. This file specifies which systems to build. Use the provided starter configuration as a template:
+
+   ```bash
+   cp config/starter-config.exs config/config.exs
+   ```
+
+   The `starter-config.exs` file includes example configurations for common hardware platforms.
+
+2. **Modify the Configuration File**
+   Open the newly created `config/config.exs` file in a text editor. Review the listed systems and customize the configuration to include only the systems you want to build. For example:
+
+   - **Delete unnecessary systems** to speed up the build process.
+   - **Add new systems** by following the format in the starter configuration.
+
+   An example configuration in `config.exs` might look like this:
+
+   ```elixir
+   config :nerves_systems,
+     systems: [
+       "nerves_system_rpi0",
+       "nerves_system_rpi3",
+       "nerves_system_bbb"
```

Comment:
This isn't right. It should be something like:

```
{"rpi0", "git@github.com:nerves-project/nerves_system_rpi0", branch: "main"},
...
```

Reply:
Oh I overlooked it. I guess we can remove this snippet from the guide since the users is expected to reference the starter configuration.